tiledb/sm/storage_manager/context.cc

```diff
@@ -49,7 +49,8 @@ Context::Context()
     : last_error_(Status::Ok())
     , storage_manager_(nullptr)
     , stats_(tdb::make_shared<stats::Stats>(HERE(), "Context"))
-    , logger_(tdb::make_shared<Logger>(HERE(), "")) {
+    , logger_(tdb::make_shared<Logger>(
+          HERE(), "Context: " + std::to_string(++logger_id_))) {
```

Comment:
FWIW, I think attaching the context address as the logger id here would be a bit more useful. Unless I missed something, the incremented integer is arbitrary based on init order, so it doesn't seem to provide additional information. Whereas having the object id in the log could be very helpful in a debugger or core dump analysis.

(that would probably require moving the logger construction from the initializer list to the constructor body -- perhaps not ideal, but don't think we're expecting anyone to subclass Context...)

Reply:
The goal of the ID is so we can filter logs from a multi-operation (multiple-context/query/readers) in order to isolate a single object's logs.

Reply:
Right, but the object id is unique. NBD, seemed like a small ergonomics improvement for the in-debugger use case, but I haven't really used the logger for other purposes. Marking resolved.